src/target/stm32f1.c

```diff
@@ -173,10 +176,21 @@ bool gd32f1_probe(target_s *target)
 	const uint16_t flash_size = signature & 0xffffU;
 	const uint16_t ram_size = signature >> 16U;
 
+	/*
+	 * GD32F303x User Manual Rev2.9, 2.3.1 Flash memory architecture:
+	 * HD and <=512 KiB CL devices have only bank0 with 2 KiB sized pages
+	 * XD and  >512 KiB CL devices also have bank1 with 4 KiB sized pages
+	 */
+	if (flash_size > 512U) {
```

Comment:
I'm a bit concerned that this isn't coordinated with `FLASH_BANK_SPLIT`, and that it might catch other GD32 parts that have more than 512kiB of flash, but don't have split banks.

Reply:
Are there such targets BMD explicitly supports? I couldn't detect them here. This STM32F1 driver expects a bank split at 512 KiB, and I don't want to refactor it much (because it incurs giant amounts of regress-testing).
GD32F450/470 are handled by STM32F4 flash driver because of their flash controller resembling F2/F4/F7. Even newest GD32H7 have a single bank of 3840 KiB in 4 KiB pages.
I try to be forward-looking and to break ideally zero targets, but looks like this project is contributor-driven w.r.t. less popular targets' driver support. I waited two months between posting Issue 1426 and submitting PR1503. Feel free to mention your target you use now (or will use soon) that needs additional handling in BMD auto-detection logic.

Reply:
Sorry, I was trying to suggest that some slight future-proofing of the code. A little bit of documentation is probably good enough.

It might be helpful to add comments in this area, and around the existing code that deals with `FLASH_BANK_SPLIT` (maybe just the `FLASH_BANK_SPLIT` definition), warning that they have to be manually checked for consistency when that code is modified.

Reply:
Either works for me; i'll leave it up to @dragonmux as the maintainer.

Reply:
Think we're fine with just adding a comment for now (use `/* XXX: <message> */` to make it stand out as a known issue) and the next person who messes with this code can make it right. If the comment references this thread then it should provide all the context required.

We're working on the assumption that you'd rather not put the work and spoons in right now into doing this "right" rather than "good enough". It probably should do the proper thing for bank splitting though.

We don't mind taking it as is (+ comment) or fixed though - that's up to what you feel comfortable doing.

Reply:
Expanded the comment a bit to set up expectations / remind of de-facto API.

Reply:
Thank you, think that resolves this thread for now w/ the expectation that it'll be revisited at some later date to make it more broadly correct.